### DISCUSSION TITLE
Update current state of Riverpod saved data

### DISCUSSION THREAD

**Discussion by pishguy** [rrousselGit/riverpod#696] [Q&A]
Since some days ago which i have this problem i can't get reply on [Discord](https://discord.com/channels/765557403865186374/765557404766830614/878317503921659914) and [Stackoverflow.com](https://stackoverflow.com/q/68859584/1830228) and i can't found any body to help me to resolve this problem

Like with this sample code which that's into `Riverpod` sample codes for updating the current state of stored data:

```dart
state = [
  for (final todo in state)
    if (todo.id == id)
      Todo(
        id: todo.id,
        completed: todo.completed,
        description: description,
      )
    else
      todo,
];
```

i try to implementing that on my class data, but it saved data as new without updating selected state, my class data:

```dart
class OrderStructure {
  final String title;
  final List<SelectedProducts> products;

  const OrderStructure(this.title, this.products);
}

class SelectedProducts {
  final int id;
  final List<SelectedProductServices> services;

  SelectedProducts(this.id, this.services);
}

class SelectedProductServices {
  final int id;
  final int count;
  final int cost;

  const SelectedProductServices(this.id, this.count, this.cost);
}
```

here i try to found product's services and updating count of that as decrement `count-1`, but after search my code cause add new service into product and couldn't update

here i want to decrement count on `service` when `product.id == productId` and `service.id == serviceId`
```dart
void decrement(int productId, int serviceId) {
  final newState = OrderStructure(state.title, [
    ...state.products,
    for (final product in state.products)
      if (product.id == productId)

        /* product.services is nested List class data*/
        for (final service in product.services)
          if (service.id == serviceId)

            /* i have problem on this lines*/
            SelectedProducts(productId, [
              ...product.services.where((s) => s.id == serviceId),
              SelectedProductServices(service.id, service.count - 1, service.cost)
            ])

            /* SelectedProducts(productId, [
              ...product.services,
              SelectedProductServices(service.id, service.count - 1, service.cost)
            ]) */

          else
            SelectedProducts(productId,product.services)
  ]);

  state = newState;
```

**Comment by smiLLe** [comment]
Where is `state` coming from?

**Comment by pishguy** [comment]
its a StateNotifierProvider

this link is Github sample code implementation:

https://github.com/pishguy/riverpod_sample

**Comment by pishguy** [comment]
@smiLLe i published source code on github
